### ISSUE TITLE
ECONNRESET

### ISSUE THREAD

**Issue by kumavis** [ssbc/patchwork#428]
fresh install

internet is behaving normally, tho i am using a vpn.

https://gist.github.com/kumavis/d8e9b45f269c6ac8125396648ad75466

**Comment by dominictarr** [comment]
long time no see! when did you last run patchwork?
one idea, maybe this is happening because now there is stuff to prioritize which pub you join, intending to avoid needlessly connecting to too many pubs (which wastes bandwidth) maybe the pubs you joined with are no longer the same?

maybe try a fresh invite code: `
wx.larpa.net:8008:@DTNmX+4SjsgZ7xyDh5xxmNtFqa6pWi5Qtw7cE8aR9TQ=.ed25519~NIZK4xDq3JDl6cfEpIyi3kGHIf2kYf7ov5ix/icPADM=`
/cc @mmckegg

**Comment by kumavis** [comment]
This seemed to work a bit better, but it crashed again with a similar error
https://gist.github.com/kumavis/d6453d0e68a593bf73b668aadb28e960

**Comment by kumavis** [comment]
```
RangeError: out of range index
    at RangeError (native)
    at fastCopy (/Users/kumavis/.nvm/versions/node/v6.3.1/lib/node_modules/ssb-patchwork/node_modules/ws/lib/Receiver.js:318:24)
    at Receiver.add (/Users/kumavis/.nvm/versions/node/v6.3.1/lib/node_modules/ssb-patchwork/node_modules/ws/lib/Receiver.js:78:3)
    at Socket.firstHandler (/Users/kumavis/.nvm/versions/node/v6.3.1/lib/node_modules/ssb-patchwork/node_modules/ws/lib/WebSocket.js:718:22)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:177:18)
    at Socket.Readable.push (_stream_readable.js:135:10)
    at TCP.onread (net.js:542:20)
```
is the one that kills it, all the other errors are just... console decoration

**Comment by kumavis** [comment]
patchwork is looking awesome and running snappy btw : )

**Comment by dominictarr** [comment]
I saw your new posts on patchwork, so did you get this working?

**Comment by kumavis** [comment]
yeah it works for a couple minutes before it crashes, and then I navigate somewhere and it crashes, though it doesnt seem deterministic

**Comment by mmckegg** [comment]
Yeah, I've been seeing this issue in patchwork for a while too. Ever since the upgrade to node v6 / electron 1.4. Something to do with websockets. 

Maybe relevant: https://github.com/websockets/ws/issues/778

Some reports have suggested that upgrading ws to a more recent version could fix. It is a dependency of [pull-ws-server](https://github.com/pull-stream/pull-ws-server) which is pinned to https://github.com/dominictarr/ws/commit/590d0ea0d62b0c533cd6f26ff0e3f31f946a80db which is still at 0.5.0 (latest is 1.1.1)

**Comment by kumavis** [comment]
Filed "Update to latest ws" https://github.com/pull-stream/pull-ws-server/issues/17

**Comment by kumavis** [comment]
@mmckegg 
bump to `ws@1.1.0` was merged into `pull-ws-server@1.9.5` 

@dominictarr added that:
> this repo is no longer necessary, it's merged into pull-ws, we can deprecate and point to pull-ws

**Comment by ahdinosaur** [comment]
hey. i'm closing this as stale due to [`v3` release](https://github.com/ssbc/patchwork/releases).

if it's still a problem, open again!
